pkg/tfgen/installation_docs_test.go

```diff
@@ -157,6 +157,57 @@ func TestWriteInstallationInstructions(t *testing.T) {
 	})
 }
 
+func TestWriteOverviewHeader(t *testing.T) {
+	t.Parallel()
+	type testCase struct {
+		// The name of the test case.
+		name     string
+		input    string
+		expected string
+	}
+
+	testCases := []testCase{
+		{
+			name:     "Writes When Content Exists",
+			input:    readTestFile(t, "write-overview-header/with-overview-text.md"),
+			expected: "## Overview\n\n",
+		},
+		{
+			name:     "Does Not Write For Empty Overview",
+			input:    readTestFile(t, "write-overview-header/empty-overview.md"),
+			expected: "",
+		},
+		{
+			name:     "Does Not Write For Empty Content",
+			input:    "",
+			expected: "",
+		},
+	}
+	for _, tt := range testCases {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			actual := writeOverviewHeader([]byte(tt.input))
+			assert.Equal(t, tt.expected, actual)
+		})
+	}
+
+	// The following mirrors the way that the result of `writeOverviewHeader` gets applied to our installation doc.
+	contextTest := testCase{
+		name:     "Does Not Write For Empty Overview With Context",
+		input:    readTestFile(t, "write-overview-header/empty-overview-with-context-input.md"),
+		expected: readTestFile(t, "write-overview-header/empty-overview-with-context-expected.md"),
+	}
+	t.Run(contextTest.name, func(t *testing.T) {
+		t.Parallel()
+		text, err := removeTitle([]byte(contextTest.input))
+		require.NoError(t, err)
+		header := writeOverviewHeader(text)
+		actual := header + string(text)
```

Comment:
This follows how the surrounding code operates here. It mirrors the context surrounding this function, so we can test on an actual input file, rather than just assert that `write/getOverview` does the right thing.